### PR TITLE
[WIP] added theme/ConfigurablePlatformTheme.php

### PR DESCRIPTION
**Pull request by dietertaotesting**

### CODE REVIEW

File: models/classes/theme/ConfigurablePlatformTheme.php [hunk 1]
```diff
@@ -222,59 +313,122 @@ public function getLabel()
...
+            static::LABEL        => $options[static::LABEL],
+            static::CONTAINER_ID => $options[static::CONTAINER_ID],
+            static::ID           => $options[static::CONTAINER_ID]
+                                    . StringUtils::camelize(StringUtils::removeSpecChars($options[static::LABEL]), true)
+        ],
+            $options
+        );
+
+        if($options[static::LOGO_URL] === static::DEFAULT_PATH) {
+            $options[static::LOGO_URL] = Template::img($this->defaultThemePath . '/logo.png', $options[static::CONTAINER_ID]);
```
Comment (chilly86) [inline models/classes/theme/ConfigurablePlatformTheme.php]: so the img and css url should look like this? is this okay like this?
taoPremium/views/img/themes/platform/jockeyewing/logo.png
taoPremium/views/css/themes/platform/jockeyewing/theme.css

File: models/classes/theme/ThemeService.php [hunk 2]
```diff
@@ -38,16 +38,25 @@ class ThemeService extends ConfigurableService {
     const OPTION_HEADLESS_PAGE = 'headless_page';
 
     /**
-     * Get the current Theme
+     * Get the current theme
+     * @return Theme
      */
-    public function getTheme()
-    {
+    public function getTheme() {
```
Comment (chilly86) [inline models/classes/theme/ThemeService.php]: it's already implemented

File: models/classes/theme/ThemeService.php [hunk 3]
```diff
@@ -38,16 +38,25 @@ class ThemeService extends ConfigurableService {
...
+        $themeId = $this->getThemeId();
+        return $this->getThemeById($themeId);
+    }
+
+
+    /**
+     * Get the id of the current theme
+     * @return mixed|string
+     */
+    public function getThemeId() {
```
Comment (chilly86) [inline models/classes/theme/ThemeService.php]: you have getCurrentThemeId() instead of this

File: models/classes/theme/ThemeService.php [hunk 4]
```diff
@@ -223,4 +232,37 @@ protected function getThemeDetailsProviders()
...
     }
+
+
+    /**
+     * Remove all themes that share a certain class name
+     *
+     * @param $className
+     * @return mixed
+     */
+    public function removeThemesByClassName($className) {
```
Comment (chilly86) [inline models/classes/theme/ThemeService.php]: this should be in an update/install script, it's not a generic something what we need

File: models/classes/theme/ThemeService.php [hunk 5]
```diff
@@ -223,4 +232,37 @@ protected function getThemeDetailsProviders()
...
+    }
+
+
+    /**
+     * Remove a theme with a certain id
+     *
+     * @param $id
+     * @return mixed
+     */
+    public function removeThemeById($id) {
```
Comment (chilly86) [inline models/classes/theme/ThemeService.php]: this should be on the interface as well with boolean return value

File: models/classes/theme/ConfigurablePlatformTheme.php [hunk 6]
```diff
@@ -222,59 +313,122 @@ public function getLabel()
...
+            $cls = get_class($this);
+            strtok($cls, '\\');
+            $options[static::CONTAINER_ID] = strtok('\\');
+        }
+        $options = array_merge([
+            static::STYLESHEET   => Template::css('tao-3.css', 'tao'),
+            static::LOGO_URL     => Template::img('tao-logo.png', 'tao'),
+            static::LABEL        => $options[static::LABEL],
+            static::CONTAINER_ID => $options[static::CONTAINER_ID],
+            static::ID           => $options[static::CONTAINER_ID]
```
Comment (chilly86) [inline models/classes/theme/ConfigurablePlatformTheme.php]: this options offset should be ID I guess

File: models/classes/theme/ConfigurablePlatformTheme.php [hunk 7]
```diff
@@ -222,59 +313,122 @@ public function getLabel()
...
+    /**
+     * Retrieve all custom translatable strings for the given keys
+     *
+     * @param array $keys
+     * @return array
+     */
+    public function getTextFromArray(array $keys = [])
+    {
+        $values = [];
+        forEach ($keys as $key) {
```
Comment (chilly86) [inline models/classes/theme/ConfigurablePlatformTheme.php]: lowercase foreach please!

File: models/classes/theme/ConfigurablePlatformTheme.php [hunk 8]
```diff
@@ -20,124 +20,224 @@
...
-    /** Theme id offset in the options. */
-    const THEME_ID     = 'id';
 
-    /** Theme prefix offset in the options. */
-    const THEME_PREFIX = 'prefix';
+    /** Theme container id key */
+    const CONTAINER_ID = 'containerId';
+    
+    /** Theme container type key */
+    const CONTAINER_TYPE = 'containerType';
```
Comment (chilly86) [inline models/classes/theme/ConfigurablePlatformTheme.php]: we don't need this option I guess.

File: models/classes/theme/ConfigurablePlatformTheme.php [hunk 9]
```diff
@@ -222,59 +313,122 @@ public function getLabel()
...
+    /**
+     * Construct the common part of the default theme part
+     *
+     * @param string $label
+     */
+    protected function setDefaultThemePath($label)
+    {
 
-        $this->label = $this->getOption(static::THEME_LABEL);
+        $this->defaultThemePath = 'themes/platform/' . StringUtils::removeSpecChars($label);
```
Comment (chilly86) [inline models/classes/theme/ConfigurablePlatformTheme.php]: please put the themes/platform/ to a class constant!

File: models/classes/theme/ConfigurablePlatformTheme.php [hunk 10]
```diff
@@ -20,124 +20,224 @@
...
     /**
-     * Get a template associated to given $id
+     * Get a template associated from a given $id
      *
      * @param string $id
      * @param string $context
-     * @return null|string
+     * @return string
      */
     public function getTemplate($id, $context = Theme::CONTEXT_BACKOFFICE)
```
Comment (chilly86) [inline models/classes/theme/ConfigurablePlatformTheme.php]: we don't use $context in this method, why do we need to have?

File: models/classes/theme/ConfigurablePlatformTheme.php (hunk 7 above)
Comment (dietertaotesting) [reply models/classes/theme/ConfigurablePlatformTheme.php]: I'm writing too much JavaScript...

File: models/classes/theme/ConfigurablePlatformTheme.php (hunk 10 above)
Comment (dietertaotesting) [reply models/classes/theme/ConfigurablePlatformTheme.php]: compat with `\oat\tao\model\theme\Theme::getTemplate`

File: models/classes/theme/ConfigurablePlatformTheme.php (hunk 1 above)
Comment (dietertaotesting) [reply models/classes/theme/ConfigurablePlatformTheme.php]: confirmed

File: models/classes/theme/ConfigurablePlatformTheme.php (hunk 6 above)
Comment (dietertaotesting) [reply models/classes/theme/ConfigurablePlatformTheme.php]: No, this code code is correct. Given a label _My Label_ and and an extension _taoFoo_it the id would be _taoFooMyLabel_

File: models/classes/theme/ConfigurablePlatformTheme.php (hunk 10 above)
Comment (chilly86) [reply models/classes/theme/ConfigurablePlatformTheme.php]: understood! :)

**Pull request merged**